go/worker/keymanager/worker.go

```diff
@@ -459,12 +479,17 @@ func (w *Worker) worker() { // nolint: gocyclo
 			}
 
 			// Forward status update to key manager runtime.
-			if err := w.updateStatus(currentStatus, currentStartedEvent); err != nil {
+			if err = w.updateStatus(currentStatus, currentStartedEvent); err != nil {
 				w.logger.Error("failed to handle status update",
 					"err", err,
 				)
 				continue
 			}
+		case <-w.initTickerCh:
+			if err = w.updateStatus(currentStatus, currentStartedEvent); err != nil {
```

Comment:
Maybe check that `currentStatus` and `currentStartedEvent` are still available?

Reply:
good idea, done :+1: